File: equipment-backend/src/test/java/by/gto/equipment/test/ReferenceResourceTest.java
// package by.gto.equipment.test;
//
// import static io.restassured.RestAssured.given;
// import static org.hamcrest.CoreMatchers.is;
//
// import by.gto.equipment.test.helpers.LogHelpers;
// import by.gto.equipment.test.lifecycle.MariaDBTestResource;
// import io.quarkus.test.common.QuarkusTestResource;
// import io.quarkus.test.junit.QuarkusTest;
// import java.util.Date;
// import java.util.Map;
// import javax.ws.rs.core.MediaType;
// import javax.ws.rs.core.Response;
// import org.eclipse.microprofile.config.Config;
// import org.eclipse.microprofile.config.ConfigProvider;
// import org.eclipse.microprofile.config.inject.ConfigProperty;
// import org.junit.jupiter.api.Test;
//
// @QuarkusTest
// @QuarkusTestResource(MariaDBTestResource.class)
// public class ReferenceResourceTest {
//     @Test
//     void testStatesEndpoint() throws Throwable {
//
//         Throwable resultException = null;
//
//         final io.restassured.response.Response response = given().when().get("/r/eqStates");
//         response.then().statusCode(Response.Status.OK.getStatusCode())
//             .contentType(MediaType.APPLICATION_JSON);
//
//
//         try {
//             final Map result = given()
//                 .when().get("/r/eqStates")
//                 .then()
//                 .statusCode(Response.Status.OK.getStatusCode())
//                 .contentType(MediaType.APPLICATION_JSON)
//                 // .body(is("Hello RESTEasy"))
//                 //                .body(matchesJson(""))
//                 .extract().as(Map.class);
//             for (Object key : result.keySet()) {
//                 sb.append(key).append(" => ").append(result.get(key)).append("\n");
//             }
//         } catch (Throwable ex) {
//             sb.append(ex);
//             resultException = ex;
//         }
//
//         LogHelpers.writeCharSequenceToFile(sb, "build/answer.txt");
//         if (resultException != null) throw resultException;
//     }
// }

File: equipment-backend/src/test/kotlin/by/gto/equipment/test/ReferenceResourceTest.kt
package by.gto.equipment.test

import by.gto.equipment.test.helpers.LogHelpers
import by.gto.equipment.test.lifecycle.MariaDBTestResource
import io.quarkus.test.common.QuarkusTestResource
import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured
import io.restassured.response.ValidatableResponse
import org.eclipse.microprofile.config.ConfigProvider
import org.junit.jupiter.api.Test
import java.util.Date
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response
import  io.restassured.RestAssured.*
import io.restassured.matcher.RestAssuredMatchers.*
import org.hamcrest.Matchers.*
import org.junit.jupiter.api.Assertions
import kotlin.collections.Map

@QuarkusTest
@QuarkusTestResource(MariaDBTestResource::class)
class ReferenceResourceTest {

    @Test
    fun logSomeInfo() {
        val sb = StringBuilder(Date().toString()).append("\n\n")
        val config = ConfigProvider.getConfig()
        for (propertyName in config.propertyNames) {
            sb.append(propertyName).append(" => ").append(config.getConfigValue(propertyName).rawValue).append("\n")
        }
        LogHelpers.writeCharSequenceToFile(sb.append("\n\n"), "build/quarkus-test-properties.log")
    }

    @Test
    @Throws(Throwable::class)
    fun testStatesEndpoint() {
        val response = RestAssured.given().`when`().get("/r/eqStates")
        val vResponse: ValidatableResponse = response.then()
        vResponse.statusCode(Response.Status.OK.statusCode)
                .contentType(MediaType.APPLICATION_JSON)
//        val body: Map<String, Any?> = vResponse.extract().path("$")
//        vResponse.body("date", `is`(""))
    }
}
